### PR TITLE
♻️ SonarCloud issues in `director-v2` and `dynamic-sidecar`

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Minor complains from SonarCloud started failing. This address them.

## Related issue/s

- https://github.com/ITISFoundation/osparc-issues/issues/675
- https://github.com/ITISFoundation/osparc-simcore/issues/4120

## How to test

- tests for these changes exist

### CODE REVIEW

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py [hunk 1]
```diff
@@ -198,7 +196,7 @@ def remap_to_compose_spec_key() -> dict[str, str]:
         # remaps from image_name as key to compose_spec key
         return {reverse_mapping[k]: v for k, v in docker_image_name_by_services.items()}
 
-    compose_spec: Optional[ComposeSpecLabel] = cast(
+    compose_spec: ComposeSpecLabel | None = cast(
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py:199]: is this cast really needed?

**Review by sanderegg — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py:199]: Yes, the `Json` type there is used to automatically decode the string to an object. This is still required.

**Review by pcrespov — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/settings.py:199]: So apparently if you cast None to an object you can do it and it is still None.
```
from typing import cast


class A:
    pass


none_cast = cast(A, None)
print(none_cast)
assert cast(A, None) is None
```

**Pull request merged**